### PR TITLE
Replaced xhrpatch with a custom ajax function

### PR DESCRIPTION
**Pull request by tomivirkki**
[Review on Reviewable](https://reviewable.io/reviews/vaadin/vaadin-grid/306)

### CODE REVIEW

**Comment by tehapo** [comment]
Rebased this branch to include latest changes from master.

**Comment by tehapo** [comment]
Fixed the single linter error that was breaking the build.

**Comment by Saulis** [comment]
I'm getting this on Chrome on the second Angular 2 example.



---

Reviewed 14 of 15 files at r1, 1 of 1 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-grid/306#-:-K7kXqvuUnjkwfmgdXfd:951146773)*

**Comment by tomivirkki** [comment]
Doesn't relate to this change. Seems to occur in master also.

---







*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-grid/306#-:-K7kZl2CIk_i-pfr4pUm:1097181621)*

**Comment by tomivirkki** [comment]
fyi: fixed the issue in master and rebased this branch

---







*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-grid/306#-:-K7kbCD_W11xtLZG3D2o:-1672088641)*

**Comment by Saulis** [comment]
:shipit: 

---

Review status: 14 of 15 files reviewed at latest revision, all discussions resolved.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/vaadin/vaadin-grid/306#-:-K7kiUQy2QmQvCi4Tqwv:1148172450)*

File: vaadin-grid.html [hunk 1]
```diff
@@ -6,6 +6,19 @@
 
 <link rel='import' href='../polymer/polymer.html'>
 <script src="vaadin-grid.min.js"></script>
+<script>
+if (!window.vaadin) {
```
Comment (manolo) [inline vaadin-grid.html]: This is an issue using the component in vaadin framework, since window.vaadin already exists

Comment (tomivirkki) [inline vaadin-grid.html]: How is it a problem? If the namespace already exists then sdm.js simply isn't imported (as it shouldn't). We don't run sdm with Vaadin Framework during Core Elements development.

Comment (manolo) [inline vaadin-grid.html]: Well we could be interested sometime in fact I debugged vaadin-grid in SDM with other GWT apps running SDM but never pure vaadin ones.
Anyway, you are right this is a very corner case.

**Pull request merged**